src/pybind/mgr/volumes/fs/nfs.py

```diff
@@ -0,0 +1,373 @@
+
+    def _save_export(self, export):
+        self.exports[self.rados_namespace].append(export)
+        conf_block = export.to_export_block()
+        self._write_raw_config(conf_block, "export-{}".format(export.export_id))
+        self._update_common_conf(export.export_id)
+
+    def create_export(self, export_type, fs_name, pseudo_path, read_only, path, cluster_id):
+        if export_type != 'fs':
+            return -errno.EINVAL,"", "Invalid FSAL. Only CephFS is supported."
+        #TODO Check if valid cluster
+        if cluster_id not in self.exports:
+            self.exports[cluster_id] = []
+
+        self.rados_namespace = cluster_id
+        if not self.check_fs(fs_name) or self.check_pseudo_path(pseudo_path):
+            return -errno.EINVAL,"", "Invalid CephFS name or export already exists"
+
+        user_id, key = self._create_user_key(cluster_id)
+        if isinstance(user_id, int):
+            return user_id, "", key
+        access_type = "RW"
+        if read_only:
+            access_type = "R"
+
+        ex_dict = {
+            'path': self.format_path(path),
+            'pseudo': self.format_path(pseudo_path),
+            'cluster_id': cluster_id,
+            'access_type': access_type,
+            'fsal': {"name": "CEPH", "user_id":cluster_id, "fs_name": fs_name, "sec_label_xattr": ""},
+            'clients': []
+            }
+
+        ex_id = self._gen_export_id()
+        export = Export.from_dict(ex_id, ex_dict)
+        export.fsal.cephx_key = key
+        self._save_export(export)
+
+        result = {
+            "bind": pseudo_path,
+            "fs": fs_name,
+            "path": path,
+            "cluster": cluster_id,
+            "mode": access_type,
+            }
+
+        return (0, json.dumps(result, indent=4), '')
+
+    def delete_export(self, ex_id):
+        raise NotImplementedError()
+
+    def make_rados_url(self, obj):
+        if self.rados_namespace:
+            return "rados://{}/{}/{}".format(self.rados_pool, self.rados_namespace, obj)
+        return "rados://{}/{}".format(self.rados_pool, obj)
+
+class NFSCluster:
+    def __init__(self, mgr, cluster_id):
+        self.cluster_id = "ganesha-%s" % cluster_id
+        self.pool_name = 'nfs-ganesha'
+        self.pool_ns = cluster_id
+        self.mgr = mgr
+
+    def create_empty_rados_obj(self):
+        common_conf = 'conf-nfs'
+        result = ''
+        with self.mgr.rados.open_ioctx(self.pool_name) as ioctx:
+            if self.pool_ns:
+                ioctx.set_namespace(self.pool_ns)
+            ioctx.write_full(common_conf, result.encode('utf-8'))
+            log.debug(
+                    "write configuration into rados object %s/%s/nfs-conf\n",
+                    self.pool_name, self.pool_ns)
+
+    def create_nfs_cluster(self, export_type, size):
+        if export_type != 'fs':
+            return -errno.EINVAL,"", "Invalid FSAL. Only CephFS is supported."
```

Comment:
does cluster create differ based on export_type?

Reply:
Yes, see Jeff's comment [here](https://github.com/ceph/ceph/pull/32297#discussion_r401788143)

Reply:
Based on Jeff's comment, maybe we create a config that applies for both RGW and cephfs use-cases? ... and then the only difference would the be FSAL export block?

Reply:
We can do that but first let's get it working with cephfs.